cdk/src/functions/enable-securityhub.ts

```diff
@@ -0,0 +1,546 @@
+/**
+ *  Copyright 2022 Amazon.com, Inc. or its affiliates. All Rights Reserved.
+ *
+ *  Licensed under the Apache License, Version 2.0 (the "License"). You may not use this file except in compliance
+ *  with the License. A copy of the License is located at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ *  or in the 'license' file accompanying this file. This file is distributed on an 'AS IS' BASIS, WITHOUT WARRANTIES
+ *  OR CONDITIONS OF ANY KIND, express or implied. See the License for the specific language governing permissions
+ *  and limitations under the License.
+ */
+
+import {
+  DeregisterDelegatedAdministratorCommand,
+  EnableAWSServiceAccessCommandOutput,
+  ListAccountsCommand,
+  ListDelegatedAdministratorsCommand,
+  Organizations,
+} from '@aws-sdk/client-organizations';
+import {
+  SecurityHub,
+  CreateFindingAggregatorCommand,
+  ListFindingAggregatorsCommand,
+  UpdateFindingAggregatorCommand,
+  DeleteFindingAggregatorCommand,
+  ResourceConflictException,
+  EnableSecurityHubCommand,
+  ListOrganizationAdminAccountsCommand,
+  EnableOrganizationAdminAccountCommand,
+  DisableOrganizationAdminAccountCommand,
+  UpdateOrganizationConfigurationCommand,
+  CreateMembersCommand,
+  ListMembersCommand,
+  DisassociateMembersCommand,
+  DeleteMembersCommand,
+  DescribeStandardsControlsCommandOutput,
+  DescribeStandardsControlsCommand,
+  GetEnabledStandardsCommand,
+  StandardsStatus,
+  DescribeStandardsCommand,
+  BatchEnableStandardsCommand,
+  BatchDisableStandardsCommand,
+  UpdateStandardsControlCommand,
+} from '@aws-sdk/client-securityhub';
+import { STS } from '@aws-sdk/client-sts';
+import { getCredsFromAssumeRole } from '../utils/assume-role';
+import { delay, throttlingBackOff } from '../utils/throttle';
+
+export async function handler(event: AWSLambda.CloudFormationCustomResourceEvent) {
+  const homeRegion = event.ResourceProperties.region;
+  const adminAccountId = event.ResourceProperties.adminAccountId;
+  const secHubCrossAccountRoleArn = event.ResourceProperties.role;
+
+  const organizationsClientManagementAccount = new Organizations({ region: 'us-east-1' });
+  const securityHubClientManagementAccount = new SecurityHub();
+
+  const stsClient = new STS();
+  const creds = await getCredsFromAssumeRole(stsClient, secHubCrossAccountRoleArn, 'EnableSecurityHub');
+  const organizationsClientAuditAccount = new Organizations({ region: 'us-east-1', credentials: creds });
+  const securityHubClientAuditAccount = new SecurityHub({ credentials: creds });
+
+  switch (event.RequestType) {
+    case 'Create':
+    case 'Update':
+      await enableOrganisationAdmin(securityHubClientManagementAccount, adminAccountId, homeRegion);
```

Comment:
maybe each function should be moved to a separate file/module. In the current form this is hard to read and test.

Reply:
done